File: Regression/Section 7 - Support Vector Regression (SVR)/svrTrial.py
# Support Vector Regression (SVR)
# import the libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

# import the dataSet 
dataSet = pd.read_csv('Position_Salaries.csv')
x = dataSet.iloc[:, 1:2].values
y = dataSet.iloc[:, 2:3].values

# Feature Scaling
from sklearn.preprocessing import StandardScaler
scX = StandardScaler()
scY = StandardScaler()
x = scX.fit_transform(x)
y = scY.fit_transform(y)

# Fit the data to regressor (SVR)
from sklearn.svm import SVR
regressor = SVR(kernel='rbf')
regressor.fit(x, y)

# Visualize the result
xGrid = np.arange(min(x), max(x), .1)
xGrid = xGrid.reshape(len(xGrid), 1)
plt.scatter(x, y, color = 'RED')
plt.plot(x, scY.inverse_transform(regressor.predict(x)), color = 'BLUE' )
plt.title('SVR Level VS Salary Prediction')
plt.xlabel('Level')
plt.ylabel('Salary')
plt.show()
#Predict the Salary for any data by replacing X in following line of code
yPredict = scY.inverse_transform(regressor.predict(scX.transform(X)))

File: Regression/Section 7 - Support Vector Regression (SVR)/svrTrial.R
# Import dataset
dataSet = read.csv('Position_Salaries.csv')
dataSet = dataSet[, 2:3]

# Fit to SVR regressor
#install.packages('e1071')
library(e1071)
regressor = svm(formula = Salary~.,
                data = dataSet,
                type = 'eps-regression')

# Visualize the result
library(ggplot2)
xGrid = seq(min(dataSet$Level), max(dataSet$Level), .1)
ggplot()+
  geom_point(aes(x = dataSet$Level, y = dataSet$Salary), colour = 'RED')+
  geom_line(aes(x = xGrid, y = predict(regressor,
                                               newdata = data.frame(Level = xGrid))), colour = 'BLUE')+
  ggtitle('Position VS salary')+
  xlab('Level')+
  ylab('Salary')

# Predicting salary of any point
yPred = predict(regressor, newdata = data.frame(Level = 5))